lib/iris/_data_manager.py

```diff
@@ -61,15 +61,28 @@ def __init__(self, data, fill_value=None, realised_dtype=None):
             integer or boolean masked data.
 
         """
+        # Initialise the instance.
+        self._fill_value = None
         self._lazy_array = None
         self._real_array = None
+        self._realised_dtype = None
+
+        # Assign the data payload to be managed.
         self.data = data
 
-        self._realised_dtype = None
+        # Set the lazy data realised dtype, if appropriate.
         self._realised_dtype_setter(realised_dtype)
 
-        # Must be set after the realised dtype.
-        self.fill_value = fill_value
+        # Set the fill-value, must be set after the realised dtype.
+        if ma.isMaskedArray(data) and \
```

Comment:
Can you please move this logic into its own method? There's a fair few lines of new code here in the `__init__`, which would probably be tidier if squirrelled away in a method. It would also continue the pattern started with `_realised_dtype_setter`.